### PR TITLE
New Resource: Data Factory Linked Service Key Vault Resource

### LINKED ISSUE #6914: Support for azurerm_data_factory_linked_service_key_vault
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Provide a key vault linked service.  This will be necessary for other linked services which want to use a key vault.

### New or Affected Resource(s)

* azurerm_data_factory_linked_service_key_vault

### Potential Terraform Configuration

```hcl
resource "azurerm_data_factory_linked_service_key_vault" "acctestlskv" {
  name                = "acctestlskvl%d"
  resource_group_name = azurerm_resource_group.test.name
  data_factory_name   = azurerm_data_factory.test.name
  base_url            = "https://acctestkv.vault.azure.net "
  annotations         = ["test1", "test2", "test3"]
  description         = "test description"
}
```

### References

* https://docs.microsoft.com/en-us/rest/api/datafactory/linkedservices/createorupdate#azurekeyvaultlinkedservice

* #6264

### PR DESCRIPTION
**Pull request by pearcec**
Includes
- Tests
- Documentation
- #6914

Fixes #6914

### CODE REVIEW

File: website/docs/r/data_factory_linked_service_key_vault.html.markdown [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+page_title: "Azure Resource Manager: azurerm_data_factory_linked_service_key_vault"
+description: |-
+  Manages a Linked Service (connection) between Key Vault and Azure Data Factory.
+---
+
+# azurerm_data_factory_linked_service_key_vault
+
+Manages a Linked Service (connection) between Key Vault and Azure Data Factory.
+
+~> **Note:** All arguments including the base_url will be stored in the raw state as plain-text. [Read more about sensitive data in state](/docs/state/sensitive-data.html).
```
[the remark this answers is not in the document]
Comment (pearcec) [reply website/docs/r/data_factory_linked_service_key_vault.html.markdown:13]: Ahh I meant to yank that.

File: azurerm/internal/services/datafactory/data_factory_linked_service_key_vault_resource.go [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+			// There's a bug in the Azure API where this is returned in lower-case
+			// BUG: https://github.com/Azure/azure-rest-api-specs/issues/5788
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"base_url": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.IsURLWithHTTPS,
+			},
```
[the remark this answers is not in the document]
Comment (pearcec) [reply azurerm/internal/services/datafactory/data_factory_linked_service_key_vault_resource.go:60]: Ah good point.  I was following the REST interface.  If someone generated the keyvault separate from an plan they easily add it.  If not they could create the keyvault in plan and then use output which forces the dependency.  Alternatively they could look it up with `data.azurerm_key_vault` if we went with the `key_vault_id`.  Which is probably what you would end up doing if you created it outside the plan.

- If we stick with this approach it is flexible
- If we use the `key_vault_id` people can still create key vaults outside of a plan if they choose by using a `data.azurerm_key_vault` 

My preference would be to stay consistent with how we generally approach this.

I am comfortable making it `key_vault_id` if you are.

**Comment by pearcec** [comment]
@tombuildsstuff fixed, take a look when you are able.

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2020-06-05 at 18 21 41

**Pull request merged**